src/main/java/github/scarsz/discordsrv/objects/threads/ChannelUpdater.java

```
@@ -0,0 +1,157 @@
+                return;
+            }
+        }
+    }
+
+    private static class UpdaterChannel {
+
+        @Getter private final GuildChannel discordChannel;
+        @Getter private final String format;
+        @Getter private final int interval;
+        private int minutesUntilRefresh;
+
+        public UpdaterChannel(GuildChannel channel, String format, int interval) {
+            this.discordChannel = channel;
+            this.format = format;
+
+            // Minimum value for the interval is 10 so we'll make sure it's above that
+            if (interval < 10) {
+                DiscordSRV.debug(Debug.CHANNEL_UPDATER, "Update interval for channel \"" + channel.getName() + "\" was below the minimum value of 10. Using 10 as the interval.");
+                this.interval = 10;
+            } else this.interval = interval;
+
+            this.minutesUntilRefresh = this.interval;
+
+        }
+
+        public void update () {
+            String message = PlaceholderUtil.replaceChannelUpdaterPlaceholders(this.format);
+            if (message.length() > 100) {
+                message = message.substring(0, 99);
+                DiscordSRV.debug(Debug.CHANNEL_UPDATER, "The new channel name for \"" + this.discordChannel.getName() + "\" was too long. Reducing it to 100 characters...");
+                if (StringUtils.isBlank(message)) {
+                    DiscordSRV.debug(Debug.CHANNEL_UPDATER, "The new channel name for `\"" + this.discordChannel.getName() + "\" was blank, skipping...");
+                    return;
+                }
+            }
+            try {
+                this.discordChannel.getManager().setName(message).queue();
+            } catch (Exception e) {
+                if (e instanceof PermissionException) {
+                    final PermissionException pe = (PermissionException) e;
+                    if (pe.getPermission() != Permission.UNKNOWN) {
```

Shouldn't completely ignore the exception when it's Unknown